instrumentation/github.com/gorilla/mux/otelmux/test/mux_test.go

```diff
@@ -23,6 +25,74 @@ import (
 	"go.opentelemetry.io/otel/trace"
 )
 
+type recordingResponseWriter struct {
+	writer     http.ResponseWriter
+	statusCode int
+	size       int
+}
+
+func (r *recordingResponseWriter) Header() http.Header {
+	return r.writer.Header()
+}
+
+func (r *recordingResponseWriter) Write(b []byte) (int, error) {
+	if r.statusCode == 0 {
+		r.statusCode = http.StatusOK
+	}
+	size, err := r.writer.Write(b)
+	r.size += size
+	return size, err
+}
+
+func (r *recordingResponseWriter) WriteHeader(statusCode int) {
+	r.statusCode = statusCode
+	r.writer.WriteHeader(statusCode)
+}
+
+func (r *recordingResponseWriter) Hijack() (net.Conn, *bufio.ReadWriter, error) {
+	hj, ok := r.writer.(http.Hijacker)
+	if !ok {
+		return nil, nil, http.ErrNotSupported
+	}
+	return hj.Hijack()
+}
+
+type mockHijacker struct {
+	http.ResponseWriter
+}
+
+func (m *mockHijacker) Hijack() (net.Conn, *bufio.ReadWriter, error) {
+	conn, rw := net.Pipe()
+	return conn, bufio.NewReadWriter(bufio.NewReader(rw), bufio.NewWriter(rw)), nil
+}
+
+type mockNonHijacker struct {
+	http.ResponseWriter
+}
+
+func TestRecordingResponseWriterHijack(t *testing.T) {
```

Comment:
Both those tests here seem wrong. You're not using the response writer provided by otelmux anywhere. So this doesn't test the runtime hijacker, it tests the one you're setting up within the test package.

Reply:
code ensures that the otelmux response writer is being used in the tests:


```
  func TestRecordingResponseWriterHijack(t *testing.T) {
    // Create a new response recorder
    recorder := httptest.NewRecorder()
    // Wrap the recorder with otelmux response writer
    otelWriter := otelmux.NewResponseWriter(recorder, nil)
    // Create a recordingResponseWriter using the otelmux response writer
    recWriter := &recordingResponseWriter{writer: otelWriter}

    // Perform the hijack operation
    conn, rw, err := recWriter.Hijack()
    require.NoError(t, err)
    assert.NotNil(t, conn)
    assert.NotNil(t, rw)

    // Close the connection
    err = conn.Close()
    require.NoError(t, err)
}

func TestRecordingResponseWriterHijackNonHijacker(t *testing.T) {
    // Create a new response recorder
    recorder := httptest.NewRecorder()
    // Wrap the recorder with otelmux response writer
    otelWriter := otelmux.NewResponseWriter(recorder, nil)
    // Create a recordingResponseWriter using the otelmux response writer
    recWriter := &recordingResponseWriter{writer: otelWriter}

    // Perform the hijack operation
    conn, rw, err := recWriter.Hijack()
    require.NoError(t, err)
    assert.NotNil(t, conn)
    assert.NotNil(t, rw)
}`